src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -34,12 +34,84 @@ internal class SemanticTokensHelpers
+                [ClassificationTypeNames.ClassName] = SemanticTokenTypes.Class,
+                [ClassificationTypeNames.RecordClassName] = SemanticTokenTypes.Class,
+                [ClassificationTypeNames.DelegateName] = SemanticTokenTypes.Class,
+                [ClassificationTypeNames.ModuleName] = SemanticTokenTypes.Class,
+
+                // No specific lsp property for both of these
+                [ClassificationTypeNames.StructName] = SemanticTokenTypes.Struct,
+                [ClassificationTypeNames.RecordStructName] = SemanticTokenTypes.Struct,
+
+                [ClassificationTypeNames.NamespaceName] = SemanticTokenTypes.Namespace,
+                [ClassificationTypeNames.EnumName] = SemanticTokenTypes.Enum,
+                [ClassificationTypeNames.InterfaceName] = SemanticTokenTypes.Interface,
+                [ClassificationTypeNames.TypeParameterName] = SemanticTokenTypes.TypeParameter,
+                [ClassificationTypeNames.ParameterName] = SemanticTokenTypes.Parameter,
+                [ClassificationTypeNames.LocalName] = SemanticTokenTypes.Variable,
+
+                // No specific lsp property for all of these
+                [ClassificationTypeNames.PropertyName] = SemanticTokenTypes.Property,
+                [ClassificationTypeNames.FieldName] = SemanticTokenTypes.Property,
+                [ClassificationTypeNames.ConstantName] = SemanticTokenTypes.Property,
+
+                // No specific lsp property for all of these
+                [ClassificationTypeNames.MethodName] = SemanticTokenTypes.Method,
+                [ClassificationTypeNames.ExtensionMethodName] = SemanticTokenTypes.Method,
+
+                [ClassificationTypeNames.EnumMemberName] = SemanticTokenTypes.EnumMember,
+                [ClassificationTypeNames.EventName] = SemanticTokenTypes.Event,
+                [ClassificationTypeNames.PreprocessorKeyword] = SemanticTokenTypes.Macro,
+
+                // in https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#standard-token-types-and-modifiers
+                [ClassificationTypeNames.LabelName] = "label",
+
+                // No specific lsp property for all of these
+                [ClassificationTypeNames.RegexComment] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexCharacterClass] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexAnchor] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexQuantifier] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexGrouping] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexAlternation] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexText] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexSelfEscapedCharacter] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexOtherEscape] = SemanticTokenTypes.Regexp,
```

Comment:
Validated the regex classifications are coming through

Reply:
Do we not miss out on the colorization of individual elements inside the regex string since we map them to the same token type?